source/html/renderer/src/host-config.ts

```diff
@@ -133,101 +166,123 @@ function parseFactTitleDefinition(obj: any): IFactTitleDefinition {
     return result;
 }
 
-export interface IFactSetConfig {
-    title: IFactTitleDefinition,
-    value: IFactTextDefinition,
-    spacing: number
+export class FactSetConfig {
+    title: FactTitleDefinition = new FactTitleDefinition();
+    value: FactTextDefinition = new FactTextDefinition();
+    spacing: number;
 }
 
-function parseFactSetConfiguration(obj: any): IFactSetConfig {
+function parseFactSetConfiguration(obj: any): FactSetConfig {
     return obj ? {
         title: parseFactTitleDefinition(obj["title"]),
         value: parseFactTextDefinition(obj["value"]),
         spacing: obj["spacing"]
     } : null;
 }
 
-export interface IShowCardActionConfig {
-    actionMode: Enums.ShowCardActionMode,
-    inlineTopMargin: number,
-    style?: Enums.ContainerStyle
+export class ShowCardActionConfig {
+    actionMode: Enums.ShowCardActionMode;
+    inlineTopMargin: number;
+    style?: Enums.ContainerStyle;
+
+    toJSON() {
+        return {
+            actionMode: Enums.ShowCardActionMode[this.actionMode],
+            inlineTopMargin: this.inlineTopMargin,
+            style: Enums.ContainerStyle[this.style]
+        }
+    }
 }
 
-function parseShowCardActionConfiguration(obj: any): IShowCardActionConfig {
-    return obj ? {
+function parseShowCardActionConfiguration(obj: any): ShowCardActionConfig {
+    return obj ? Object.assign(new ShowCardActionConfig(), {
         actionMode: Utils.parseHostConfigEnum(Enums.ShowCardActionMode, obj["actionMode"], Enums.ShowCardActionMode.Inline),
         inlineTopMargin: obj["inlineTopMargin"],
         style: Utils.parseHostConfigEnum(Enums.ContainerStyle, obj["style"], Enums.ContainerStyle.Emphasis)
-    } : null;
+    }) : null;
 }
 
-export interface IActionsConfig {
-    maxActions: number,
-    spacing: Enums.Spacing,
-    buttonSpacing: number,
-    showCard: IShowCardActionConfig,
-    actionsOrientation: Enums.Orientation,
-    actionAlignment: Enums.ActionAlignment
+export class ActionsConfig {
+    maxActions: number;
+    spacing: Enums.Spacing;
+    buttonSpacing: number;
+    showCard: ShowCardActionConfig = new ShowCardActionConfig();
+    preExpandSingleShowCardAction?: boolean;
+    actionsOrientation: Enums.Orientation;
+    actionAlignment: Enums.ActionAlignment;
+
+    toJSON() {
+        return {
+            maxActions: this.maxActions,
+            spacing: Enums.Spacing[this.spacing],
+            buttonSpacing: this.buttonSpacing,
+            showCard: this.showCard,
+            preExpandSingleShowCardAction: this.preExpandSingleShowCardAction,
+            actionsOrientation: Enums.Orientation[this.actionsOrientation],
+            actionAlignment: Enums.ActionAlignment[this.actionAlignment]
+        }
+    }
 }
 
-function parseActionsConfiguration(obj: any): IActionsConfig {
-    return obj ? {
+function parseActionsConfiguration(obj: any): ActionsConfig {
```

Comment:
Since ActionsConfig is now a class, this method should be moved to the class and should be renamed into simply "parse"